tests/models/test_model.py

```diff
@@ -114,6 +116,25 @@ def test_model_log():
         assert Version(loaded_model.mlflow_version) == Version(mlflow.version.VERSION)
 
 
+def test_model_log_existing_full_directory_fails():
+    with TempDir(chdr=True) as tmp:
+        sig = ModelSignature(
+            inputs=Schema([ColSpec("integer", "x"), ColSpec("integer", "y")]),
+            outputs=Schema([ColSpec(name=None, type="double")]),
+        )
+        input_example = {"x": 1, "y": 2}
+
+        experiment_id = mlflow.create_experiment("test")
+
+        model_dir = "some/path"
+        os.makedirs(model_dir)
+        with open(os.path.join(model_dir, "foo.txt"), "wt") as f:
+            f.write("dummy content")
+        with mlflow.start_run(experiment_id=experiment_id):
+            with pytest.raises(MlflowException, match="already exists and is not empty"):
+                Model.log("some/path", TestFlavor, signature=sig, input_example=input_example)
```

Comment:
I think we need to call `save_model` because `Model.log` calls `save_model` with an empty directory.

https://github.com/mlflow/mlflow/blob/dc1b21821e564c7985c0df318909ba84a90ec0f1/mlflow/models/model.py#L291

Reply:
I think we can add the following test in `tests/pyfunc/test_model_export_with_class_and_artifacts.py` to verify the behavior of `_validate_and_prepare_target_save_path`

```python
@pytest.mark.large
def test_model_log_existing_full_directory_fails(main_scoped_model_class, tmp_path):
    tmp_path.joinpath("foo.txt").write_text("dummy")
    with pytest.raises(MlflowException, match="already exists and is not empty"):
        mlflow.pyfunc.save_model(
            path=tmp_path,
            python_model=main_scoped_model_class(lambda m, df: m.predict(df)),
        )
```

Reply:
yes, this file is much better, thanks! will update

Reply:
moved + simpler test, thanks!